pkg/server/api/bundle/v1/service.go

```diff
@@ -518,3 +569,50 @@ func applyBundleMask(b *types.Bundle, mask *types.BundleMask) {
 		b.JwtAuthorities = nil
 	}
 }
+
+func fieldsFromBundleProto(proto *types.Bundle, inputMask *types.BundleMask) logrus.Fields {
+	fields := logrus.Fields{
+		"trust_domain": proto.TrustDomain,
+	}
+
+	if inputMask == nil || inputMask.RefreshHint {
+		fields["refresh_hint"] = proto.RefreshHint
+	}
+
+	if inputMask == nil || inputMask.SequenceNumber {
+		fields["sequence_number"] = proto.SequenceNumber
+	}
+
+	if inputMask == nil || inputMask.JwtAuthorities {
+		for k, v := range fieldsFromJwtAuthoritiesProto(proto.JwtAuthorities) {
+			fields[k] = v
+		}
+	}
+
+	if inputMask == nil || inputMask.X509Authorities {
+		for k, v := range fieldsFromX509AuthoritiesProto(proto.X509Authorities) {
+			fields[k] = v
+		}
+	}
+	return fields
+}
+
+func fieldsFromJwtAuthoritiesProto(jwtAuthorities []*types.JWTKey) logrus.Fields {
+	fields := logrus.Fields{}
+	for i, jwtAuthoriry := range jwtAuthorities {
+		fields[fmt.Sprintf("jwt_authority_expires_at_%d", i)] = jwtAuthoriry.ExpiresAt
+		fields[fmt.Sprintf("jwt_authority_key_id_%d", i)] = jwtAuthoriry.KeyId
+		fields[fmt.Sprintf("jwt_authority_public_key_%d", i)] = api.HashByte(jwtAuthoriry.PublicKey)
+	}
+
+	return fields
+}
+
+func fieldsFromX509AuthoritiesProto(x509Authorities []*types.X509Certificate) logrus.Fields {
+	fields := logrus.Fields{}
+	for i, x509Authority := range x509Authorities {
+		fields[fmt.Sprintf("x509_authorities_asn1_%d", i)] = api.HashByte(x509Authority.Asn1)
```

Comment:
As the value here isn't actually the ASN.1 representation of the X.509 authority, could we use a more precise key name? Maybe `x509_authorities_asn1_sha256.%d`?